sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -59,4 +60,9 @@ object DataSourceUtils {
     val name = path.getName
     !(name.startsWith("_") || name.startsWith("."))
   }
+
+  def calcDataSize(properties: Map[String, String], sizeInBytes: BigInt): BigInt = {
+    val factor = NumberUtils.toInt(properties.get("deserFactor").getOrElse("1"), 1)
```

Comment:
As documented, this factor should be a float, and also I think we can use scala type conversions to simplify the codes(without introducing NumberUtils)
```
hive.stats.deserialization.factor
Default Value:
Hive 0.13 to 2.x.x: (float) 1.0
Hive 3.0.0 and later: (float) 10.0
Added In: Hive 0.13 with HIVE-5921
Default value changed from 1.0 to 10.0 in Hive 3.0
```